substrate/frame/examples/multi-block-migrations/src/lib.rs

```diff
@@ -0,0 +1,90 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+//! # Multi-Block Migrations Example Pallet
+//!
+//! This pallet serves as a minimal example of a pallet that uses the [Multi-Block Migrations
+//! Framework](frame_support::migrations). You can observe how to configure it in a runtime in the
+//! `kitchensink-runtime` crate.
+//!
+//! ## Introduction and Purpose
+//!
+//! The primary purpose of this pallet is to demonstrate the concept of Multi-Block Migrations in
+//! Substrate. It showcases the migration of values from in the
+//! [`MyMap`](`pallet::MyMap`) storage map a `u32` to a `u64` data type using the
+//! [`SteppedMigration`](`frame_support::migrations::SteppedMigration`) implementation from the
+//! [`migrations::v1`] module.
+//!
+//! The [`MyMap`](`pallet::MyMap`) storage item is defined in this `pallet`, and is
+//! aliased to [`old::MyMap`](`migrations::v1::old::MyMap`) in the [`migrations::v1`]
+//! module.
+//!
+//! ## How to Read the Documentation
+//!
+//! To access and navigate this documentation in your browser, use the following command:
+//!
+//! - `cargo doc --package pallet-example-mbm --open`
```

Comment:
Maybe better to link to the parity.io docs?

Reply:
The pallet is not yet published, so wont work yet.